code/core/src/core-server/build-dev.ts

```diff
@@ -151,15 +151,15 @@ export async function buildDevStandalone(
   // Load second pass: all presets are applied in order
   presets = await loadAllPresets({
     corePresets: [
-      require.resolve('@storybook/core-server/dist/presets/common-preset'),
+      require.resolve('@storybook/core/core-server/presets/common-preset'),
```

Comment:
The difference is that this:
`require.resolve('@storybook/core/core-server/presets/common-preset')`
...returns a path with extension. It cannot resolve a directory that way, it has to be a file.

The call throws if the file cannot be found.
`join(dirname(require.resolve('@storybook/core/package.json')), 'path')`
Will always return a path. even if the path does not exist, and it will not try to resolve the path any further.
It won't try to find a `index.js` file.

Reason to do this:
- You want to resolve a directory
- You want to resolve a ESM file `require.resolve` can not do!, but if you pass the location of the file to something that will resolve ESM, it can resolve it correctly.
- You want to find a place on the `fs` but it might not be available yet (`require.resolve' throws when it can't find it)
- every `require-resolve` blocks the main thread. Calling that `resolve` only once, for the known base-path, then add sub-paths to it using `join` is better for performance.

Reason to do the `require.resolve()` with the full path:
- Ensure the file is resolvable
- Find the CJS version of the file

Reply:
Thank you for the clarification! 🙏